### ISSUE TITLE
Implement Limited Liability: Cap Loss Payments at Available Equity

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#199]
## Description

The current financial model violates the principle of limited liability by allowing equity to become negative when large losses occur. Companies should not be able to pay more than their available equity - once equity reaches zero, the company becomes insolvent and cannot make further payments.

This bug affects the core financial calculations and can lead to:
- Unrealistic negative equity values
- Extremely large negative growth rates (below -100% or -1.0)
- Incorrect simulation results that don't reflect real-world business constraints

## Context

**Project Area:** Core Financial Model
**Complexity:** 7/10 (Architectural change affecting multiple components)
**Type:** Bug Fix (Breaking Change)

This issue affects the fundamental accounting model in the `WidgetManufacturer` class and all downstream analyses. The current implementation allows liabilities to exceed assets, which is inconsistent with corporate limited liability principles.

## Current Behavior

When a company experiences a loss larger than available equity:

1. **Payment Processing** (`manufacturer.py:1794-1846`):
   - Company attempts to pay full loss amount
   - If loss > liquid assets, all liquid assets are depleted
   - A liability is created for the unpayable shortfall
   - This results in `Liabilities > Assets`, creating negative equity

2. **Example from test suite**:
   ```python
   # Assets: $100K, Loss: $500K
   manufacturer.process_uninsured_claim(500_000, immediate_payment=True)
   assert manufacturer.total_assets == 0
   assert manufacturer.equity == -400_000  # NEGATIVE EQUITY!
   ```

3. **Growth Rate Issues** (`ergodic_analyzer.py:654-782`):
   - Small positive equity → very large negative growth rates
   - Negative equity → returns `-inf`
   - No floor at -100% (-1.0) as expected with limited liability

## Expected Behavior (Limited Liability)

With proper limited liability implementation:

1. **Loss Payment Capping**:
   - Create liability for full loss amount
   - Cap actual payment at available equity
   - When equity reaches $0, mark unpayable debt
   - Trigger insolvency proceedings

2. **Insolvency Accounting**:
   - Follow standard accounting practices for insolvency
   - Mark company as insolvent when equity ≤ 0
   - Prevent further operations after insolvency
   - Properly record unpayable liabilities

3. **Growth Rate Floor**:
   - Natural floor at -100% (-1.0) due to equity capping
   - Explicit safety check in `calculate_time_average_growth()`
   - Maximum loss = starting timeperiod equity (100% loss)

### Affected Files

**Core Implementation:**
1. **`ergodic_insurance/manufacturer.py`** (primary changes)
   - `process_uninsured_claim()` - Cap payment at available equity
   - `process_insurance_claim()` - Cap payment at available equity
   - `pay_claim_liabilities()` - Cap scheduled payments at available equity
   - `check_solvency()` - Enhance insolvency detection (already exists)
   - NEW: `handle_insolvency()` - Implement standard insolvency accounting

2. **`ergodic_insurance/ergodic_analyzer.py`**
   - `calculate_time_average_growth()` - Add explicit -1.0 floor as safety check

3. **`ergodic_insurance/simulation.py`**
   - `run()` - Verify insolvency handling works correctly with new caps

**Test Updates Required:**
- `test_manufacturer.py`
- `test_manufacturer_methods.py`
- `test_financial_integration.py`
- `test_roe_insurance.py`
- Any tests expecting negative equity

**Documentation:**
- Update docstrings for modified methods
- Document the insolvency accounting approach

## Requirements

1. **Limited Liability Enforcement:**
   - [ ] Loss payments must never exceed available equity
   - [ ] Equity must never go below $0
   - [ ] Unpayable debts must be properly recorded as liabilities

2. **Insolvency Accounting:**
   - [ ] Implement standard insolvency accounting procedures
   - [ ] Mark company as insolvent when equity ≤ $0
   - [ ] Prevent further business operations after insolvency
   - [ ] Maintain accounting equation: Assets = Liabilities + Equity

3. **Growth Rate Protection:**
   - [ ] Implement explicit -1.0 floor in `calculate_time_average_growth()`
   - [ ] Verify floor is enforced in all growth rate calculations
   - [ ] Document that maximum loss = 100% of equity

4. **Testing & Validation:**
   - [ ] Update all tests expecting negative equity
   - [ ] Add new tests for payment capping behavior
   - [ ] Add tests for insolvency accounting
   - [ ] Verify growth rate floor is never violated

### Phase 1: Core Payment Capping

1. **Modify `process_uninsured_claim()` in `manufacturer.py`**:
   ```python
   # When immediate_payment=True:
   # 1. Calculate maximum_payable = min(claim_amount, self.equity)
   # 2. Pay maximum_payable (reduce assets/equity)
   # 3. Create liability for the full claim_amount
   # 4. If claim_amount > maximum_payable, mark unpayable portion
   # 5. Check if equity <= 0, trigger insolvency
   ```

2. **Modify `process_insurance_claim()` in `manufacturer.py`**:
   ```python
   # For company_payment portion:
   # 1. Cap company_payment at available equity
   # 2. If capped, trigger insolvency check
   # 3. Record the unpayable portion as a liability
   ```

3. **Modify `pay_claim_liabilities()` in `manufacturer.py`**:
   ```python
   # For scheduled payments:
   # 1. Calculate total_scheduled = sum of all scheduled payments
   # 2. Cap total_payment = min(total_scheduled, available_equity)
   # 3. If capped, trigger insolvency
   # 4. Allocate capped payment proportionally to claims
   ```

### Phase 2: Insolvency Accounting

4. **Create `handle_insolvency()` method in `manufacturer.py`**:
   ```python
   def handle_insolvency(self) -> None:
       """Implement standard insolvency accounting.

       When equity <= 0:
       - Set is_ruined = True
       - Freeze all operations
       - Record unpayable liabilities, something similar to:
         - Debit liabilities, credit income as "Gain on Discharge of Debt" (discharge the debt)
         - Debit retained earnings, credit equity (clear negative equity)
       - Maintain the accounting equation
       """
   ```
   Ask me clarifying questions to clarify any implementation details.

5. **Enhance `check_solvency()` in `manufacturer.py`**:
   ```python
   # Add assertion: equity should never be < 0
   # If equity == 0 within tolerance 1e-2, call handle_insolvency()
   # Return False if insolvent
   ```

### Phase 3: Growth Rate Protection

6. **Add floor to `calculate_time_average_growth()` in `ergodic_analyzer.py`**:
   ```python
   # After calculating growth_rate:
   # growth_rate = max(growth_rate, -1.0)  # Floor at -100%
   # Add a comment explaining the limited liability constraint
   ```

### Phase 4: Testing & Documentation

7. **Update test suite**:
   - Fix tests in `test_manufacturer_methods.py` expecting negative equity
   - Add test for payment capping at equity
   - Add test for insolvency accounting
   - Add test for growth rate floor
   - Verify `test_financial_integration.py` passes

8. **Update documentation**:
   - Document limited liability implementation
   - Update docstrings for modified methods
   - Add notes about breaking change

9. **Verify integration**:
   - Run full test suite
   - Run sample simulations to verify behavior
   - Check that Monte Carlo results are realistic

## Acceptance Criteria

- [ ] Equity never goes below $0 in any simulation
- [ ] Loss payments are capped at available equity
- [ ] Unpayable debts are properly recorded as liabilities
- [ ] Growth rates never fall below -1.0 (-100%)
- [ ] Insolvency is properly detected and handled according to standard insolvency accounting procedures (reflecting any reasonable simplifications in our model)
- [ ] All existing tests pass (after updates)
- [ ] New tests validate limited liability behavior
- [ ] Documentation reflects new accounting model
- [ ] Test coverage remains above 80%

## Research Notes

Standard accounting treatment for corporate insolvency:
1. When equity reaches zero, company is insolvent
2. Liabilities can exceed assets temporarily during insolvency proceedings
3. In limited liability entities, shareholders cannot be forced to pay beyond their equity
4. Unpayable liabilities remain on books until bankruptcy/reorganization
5. Company ceases normal operations when insolvent
